src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs

```diff
@@ -19,14 +17,21 @@ internal abstract partial class AbstractFullyQualifyCodeFixProvider : CodeFixPro
             public readonly int Weight;
             public readonly IReadOnlyList<string> NameParts;
 
-            public SymbolResult(INamespaceOrTypeSymbol symbol, int weight)
+            /// <summary>
+            /// Whether the original symbol at the time of creation was a type. This is unchanged by
+            /// the <see cref="WithSymbol(INamespaceOrTypeSymbol)"/> method and doesn't affect comparisons.
+            /// </summary>
+            public readonly bool OriginalSymbolIsType;
+
+            public SymbolResult(INamespaceOrTypeSymbol symbol, int weight, bool originalSymbolIsType)
             {
                 Symbol = symbol;
                 Weight = weight;
                 NameParts = INamespaceOrTypeSymbolExtensions.GetNameParts(symbol);
+                OriginalSymbolIsType = originalSymbolIsType;
```

Comment:
looks odd seeing this value passed in.  can't it be inerred from 'symbol' instead?

Reply:
looking through the PR, i think the answer is: yes.

Reply:
Sadly no, Symbol is reused to represent the containing symbol, [here](https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.cs#L315) so by the time we need to use it `Symbol.IsType` will always be `false`.

Reply:
:'(   ok :)

Reply:
alternatively, could we jus tpass that symbol along?

Reply:
Yeah, can pass around both symbols if you think its better.

Reply:
I think i would personally prefer that.